src/v2.ts

```diff
@@ -0,0 +1,153 @@
+    !!(cloudFunction?.__trigger?.httpsTrigger) &&
+    // @ts-ignore
+    cloudFunction?.__trigger?.labels?.['deployment-callable'] !== 'true'
+  ) {
+    throw new Error(
+      'Wrap function is only available for `onCall` HTTP functions, not `onRequest`.'
+    );
+  }
+
+  if (!cloudFunction.run) {
+    throw new Error(
+      'This library can only be used with functions written with firebase-functions v3.20.0 and above'
+    );
+  }
+
+  if (cloudFunction?.__endpoint?.platform !== 'gcfv2') {
+    throw new Error('This function can only wrap V2 CloudFunctions.');
+  }
+
+  return (cloudEvent: CloudEvent) => cloudFunction.run(cloudEvent);
+}
+
+/**
+ * @param cloudFunction Populates default values of the CloudEvent
+ * @param {Partial<CloudEvent>} cloudEventOverride Used to override CloudEvent params.
+ * @return {CloudEvent} Generated Mock CloudEvent
+ */
+export const createMockCloudEvent = <FunctionType, EventType>(
+  cloudFunction: CloudFunction<FunctionType>,
+  cloudEventOverride?: Partial<CloudEvent>): CloudEvent<EventType> => {
+  return {
+    ...createCloudEventWithDefaultValues<FunctionType, EventType>(cloudFunction),
+    ...cloudEventOverride,
+  } as CloudEvent<EventType>;
+};
+
+/** @internal */
+
+const getCloudEventSource =
+  <T>(cloudFunction: CloudFunction<T>): string => {
+  const projectId = '__PROJECT_ID__';
+  const type = getCloudEventType<T>(cloudFunction);
+  switch (type) {
+      case 'google.cloud.storage.object.v1.archived': // Fall-through intended
+      case 'google.cloud.storage.object.v1.deleted': // Fall-through intended
+      case 'google.cloud.storage.object.v1.finalized': // Fall-through intended
+      case 'google.cloud.storage.object.v1.metadataUpdated':
+        const bucketId = cloudFunction?.__endpoint?.eventTrigger?.eventFilters?.bucket || '';
+        return `//storage.googleapis.com/projects/_/buckets/${bucketId}`;
+      case 'google.cloud.pubsub.topic.v1.messagePublished':
+        const topicId = cloudFunction?.__endpoint?.eventTrigger?.eventFilters?.topic || '';
+        return `//pubsub.googleapis.com/projects/${projectId}/topics/${topicId}`;
+        return cloudFunction?.__endpoint?.eventTrigger?.eventFilters?.topic || '';
+      case 'google.firebase.firebasealerts.alerts.v1.published':
+        return `//firebasealerts.googleapis.com/projects/${projectId}`;
+      default:
+        return '';
+    }
+  };
+
+const getCloudEventSubject =
+  <T>(cloudFunction: CloudFunction<T>, type: string): string => {
+    switch (type) {
+      case 'google.cloud.storage.object.v1.archived': // Fall-through intended
+      case 'google.cloud.storage.object.v1.deleted': // Fall-through intended
+      case 'google.cloud.storage.object.v1.finalized': // Fall-through intended
+      case 'google.cloud.storage.object.v1.metadataUpdated':
+        return `objects/__STORAGE_FILENAME__`;
+      case 'google.firebase.firebasealerts.alerts.v1.published': // FirebaseAlerts do not contain a subject
```

Comment:
I thought firebase alerts optionally had a subject for the app ID?

Reply:
There isn't a subject, but there is a `alertType` and `appId`.

https://github.com/googleapis/google-cloudevents/blob/main/proto/google/events/firebase/firebasealerts/v1/events.proto#L39-L40